src/data/matrix/basic.lean

```diff
@@ -31,16 +31,52 @@ variables {m' : o → Type*} [∀ i, fintype (m' i)]
 variables {n' : o → Type*} [∀ i, fintype (n' i)]
 variables {R : Type*} {S : Type*} {α : Type v} {β : Type w}
 
+/-- `equiv.perm.to_matrix` converts a permutation `σ : equiv.perm m` to a matrix
+    containing ones and zeros. This definition is then used to define the Prop `matrix.is_perm`.
+    `equiv.perm.to_prequiv_to_matrix_eq_to_matrix` proves that
+    `σ.to_pequiv.to_matrix = σ.to_matrix` in `pequiv.lean`.-/
+def equiv.perm.to_matrix [decidable_eq m] [has_zero α] [has_one α] (σ : equiv.perm m) :
+matrix m m α
+| i j := if σ i = j then 1 else 0
+
```

Comment:
I'm pretty sure we don't want this definition, we already have `equiv.to_pequiv.to_matrix` and the lemma `pequiv.equiv_to_pequiv_to_matrix` which unfolds it to this.

At any rate, if we kept it, it should go at the end of `matrix/pequiv.lean`

Reply:
I will delete or move them when there is a final decision.

Reply:
It will be easier to make the decision when it's in the same file as the similar definitions that already exist.

Reply:
I think `pequiv.equiv_to_pequiv_to_matrix` defined permutation matrices in an indirect way btw, as there is no reason to first convert an `equiv` to a `pequiv`.

Reply:
When I get time, I will write more things about permutation matrices. commit a new file perhaps.

Reply:
> as there is no reason to first convert an `equiv` to a `pequiv`.

There's often a very good reason to do this. If you have separate maps from `equiv -> pequiv` and `pequiv -> matrix`, then you only have two maps to write lemmas about.
If you have a third map `equiv -> matrix`, you now need to make a copy of all the lemmas that apply to one or both of the other two maps.

One way to get out of this is to mark the `e.to_matrix = e.to_pequiv.to_matrix` lemma as `simp`, so that you can still write `e.to_matrix` as a shorthand, but the simplifier rewrites it to the longer form so that it can apply the more specific lemmas. I think that's close to what you're doing here.

It's also worth noting that you can define `perm.to_matrix` as `(1 : matrix m m α).minor σ id`, which is likely to get you more lemmas for free, and I think is definitionally identical to what you have now.